### PR TITLE
DO NOT MERGE pin fftw

### LINKED ISSUE #303: pin fftw
We should probably be pinning fftw. This dependence is very common.

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds a migrator for FFTW to ensure that all of conda-forge is on 3.3.8.

Once it is done, we should add the pin to the conda-forge-pinnings and also put in a pin run as build of `x.x`. It appears that FFTW is pretty ABI backwards compatible. 

https://abi-laboratory.pro/index.php?view=timeline&l=fftw

The other option would be to migrate to some early version to build against and then let people use w/e version they would like after that. This would be a more numpy-like setup.

Also it appears that FFTW is not using a run_export for non-mpi builds. Once we have a global pin, I suppose it could.

One possibly experimental part of this PR is that no pin for FFTW currently exists. IDK if this will confuse our bots or not.

Closes #303

### CODE REVIEW

**Comment by CJ-Wright** [comment]
Can you add the pin run as build as well?https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/271/files See

**Comment by beckermr** [comment]
Done. I didn't know that was possible!

**Comment by CJ-Wright** [comment]
@mariusvniekerk's variant merging stuff is pure magic.

**Comment by beckermr** [comment]
@conda-forge/core @conda-forge/staged-recipes Do you all have any opinions here or shall we go ahead with this version of the pinnings?

**Comment by CJ-Wright** [comment]
ABI lab seems to think that bug fixes are mostly stable https://abi-laboratory.pro/index.php?view=timeline&l=fftw so this is a green light from me. I'll wait until 12:30pm eastern for other people to comment otherwise I'll merge. (Feel free to merge without me though)

**Comment by CJ-Wright** [comment]
This has an impact on 50 packages.

File: recipe/migrations/fftw-3.3.8.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  kind:
+    version
+  # Only change the migration_number if the bot messes up,
+  # changing this will cause a complete rerun of the migration
+  migration_number:
+    1
+
+pin_run_as_build:
+  fftw:
+    max_pin: x.x
```
Comment (isuruf) [inline recipe/migrations/fftw-3.3.8.yaml]: Looking at https://abi-laboratory.pro/index.php?view=timeline&l=fftw this should be `x`

File: recipe/migrations/fftw-3.3.8.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+fftw:
+  - 3.3.8
```
Comment (isuruf) [inline recipe/migrations/fftw-3.3.8.yaml]: This should be `3.3` for `x.x` and `3` for `x`.

**Comment by CJ-Wright** [comment]
Do you want a reversion? Or do you want to correct this in master? Or do you want to put up an update PR?

**Comment by isuruf** [comment]
I don't think we need a migration. What we need is a pinning update and `repodata-patches` update for major version.

**Comment by isuruf** [comment]
See https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/305

**Pull request merged**